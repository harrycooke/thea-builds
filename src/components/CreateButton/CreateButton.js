import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import {
  Hidden,
  Fab,
} from "@material-ui/core";

//import AddIcon from "@material-ui/icons/Add";
import CreateIcon from "@material-ui/icons/Create";

import { withStyles } from "@material-ui/core/styles";



const styles = theme => ({
  fab: {
    position: 'absolute',
    bottom: theme.spacing(2),
    right: theme.spacing(3.5),
    zIndex: 9
  },
  extendedIcon: {
    marginRight: theme.spacing(1),
  },
});


const AdapterLink = React.forwardRef((props, ref) => <Link innerRef={ref} {...props} />);


export class CreateButton extends React.Component {

  componentDidMount() {
    const { id, user_type } = this.props.auth;
    const userData = {
      userId: id,
      userRole: user_type
    }
  }

    

  render() {
    const { classes } = this.props;

    return (
      <div>  
      {
        this.props.auth.user_type === "PCP" && (
          <div>
            <Hidden smUp>
              <Fab color="primary" aria-label="Create Case" component={AdapterLink} to="/create-new-case" variant="extended" className={classes.fab}>
                <CreateIcon className={classes.extendedIcon}/>
                Create a Case
              </Fab>
            </Hidden>
          </div>
        )
      }
      </div>
    );
  }
};

CreateButton.propTypes = {
  auth: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth.userdata,
});

export default connect(
  mapStateToProps,
  { Link }
)(withStyles(styles)(CreateButton));

