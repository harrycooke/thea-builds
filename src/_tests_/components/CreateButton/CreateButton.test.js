import React from 'react';
import ReactDOM from'react-dom';
import TestRenderer from 'react-test-renderer'; // ES6
import sinon from 'sinon';
import {shallow, configure, mount, render} from 'enzyme';

import {CreateButton} from '../../../components/CreateButton/CreateButton';

describe('CreateButton', () => {
    // beforeAll(() => {
    //     this.commonProps = {};
    // });

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(CreateButton, div);
    });

    it('renders React components, traversing a ReactElement tree and represents as JSON object', () => {
        const renderer = TestRenderer.create(CreateButton);
        renderer.toJSON();
      });    

    describe('CreateButton', () => {
        it('should render correctly in "debug" mode', () => {
          const component = (CreateButton);
          expect(component).toMatchSnapshot();
        });
      });

    // it('calls funcName handleSaveDraft', () => {
    //     sinon.spy(CreateButton.prototype, 'handleSaveDraft');
    //     const wrapper = mount(<CreateButton />);
    //     expect(CreateButton.prototype.componentDidMount).to.have.property('callCount', 1);
    //   });
    
    //   it('allows us to set props', () => {
    //     const wrapper = mount(<CreateButton bar="baz" />);
    //     expect(wrapper.props().bar).to.equal('baz');
    //     wrapper.setProps({ bar: 'foo' });
    //     expect(wrapper.props().bar).to.equal('foo');
    //   });
    
    //   it('simulates click events', () => {
    //     const onButtonClick = sinon.spy();
    //     const wrapper = mount((
    //       <CreateButton onButtonClick={onButtonClick} />
    //     ));
    //     wrapper.find('button').simulate('click');
    //     expect(onButtonClick).to.have.property('callCount', 1);
    //   });
});