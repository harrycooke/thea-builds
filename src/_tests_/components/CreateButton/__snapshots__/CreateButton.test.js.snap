// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateButton CreateButton should render correctly in "debug" mode 1`] = `[Function]`;
